agent/app/agent.go

```diff
@@ -479,13 +482,18 @@ func (agent *ecsAgent) registerContainerInstance(
 		tags = mergeTags(tags, ec2Tags)
 	}
 
+	if agent.registrationToken == "" {
+		agent.registrationToken = uuid.New()
+		stateManager.Save()
```

Comment:
Since we no more need to save registrationToken in the state file, we do not need to call stateManager.Save() here, considering at the end of the this 'registerContainerInstance' function, we already call stateManager.Save()  to save all saveable options.